frmts/heif/heifdrivercore.cpp

```diff
@@ -86,6 +86,9 @@ void HEIFDriverSetCommonMetadata(GDALDriver *poDriver)
 
     poDriver->pfnIdentify = HEIFDriverIdentifySimplified;
     poDriver->SetMetadataItem(GDAL_DCAP_OPEN, "YES");
+#ifdef LIBHEIF_SUPPORTS_TILES
+    poDriver->SetMetadataItem("HEIF_SUPPORTS_TILES", "YES");
```

Comment:
@rouault Should this have been in the "HEIF" metadata domain?

I'm doing a follow-up PR to make sure libheif supports what we are testing (and to provide info for debugging), and noted that you used this in the AVIF vs HEIF check. I can do the same here if preferred.

Reply:
We don't have a very strict policy on this. I'd say that you report metadata items under the non-default metadata domain for things that are so technical that they are mostly for internal GDAL use (autotest typically).